### ISSUE TITLE
Dynamic filters feedback: more types

### ISSUE THREAD

**Issue by TobiasKampAdmin** [avo-hq/avo#2946]
### Context
We updated Avo to version 3.10 and just love the new [dynamic filters](https://docs.avohq.io/3.0/dynamic-filters.html) 😍
Thanks soo much for this 🙏
Since you asked for feedback in the [release video](https://avohq.io/releases/3.10), we would like to share some thoughts. 
After giving it a spin, we are actually very enthusiastic and are just missing some functions or information about types. 

#### Types
In the release video you mentioned there are 2 types (`text` and `date`) to use when you define the dynamic_filters yourself like in the snippet below. But in the documentation it is confusing that other typed like `boolean` or `select` are also possible: [ref](https://docs.avohq.io/3.0/dynamic-filters.html#type).

```ruby
  self.dynamic_filters = {
    filters: [
      {
        label: "Firstname (custom)",
        conditions:{
          case_sensitive: "Case sensitsive",
          case_insensitive: "Case insensitve"
        }.invert,
        type: :text,  # what are the options here and how to configure?
        icon: "arrow-up-tray",
        query: -> {
          case filter_param.condition.to_sym
          when :case_sensitive
            query.where("firstname LIKE ?", "%#{filter_param.value}%")
          when :case_insensitive
            query.where("firstname ILIKE ?", "%#{filter_param.value}%")
          end 
        }
      }
    ]
  }
```

Could you provide some more information on what the options are at the moment?
When all of these options would be possible, that would be perfect for us!

If this is or will become possible, we would like some more information/documentation about the has_many, select and tags types. How to load the possible values, enforce these suggestions... 
Basically all you can do in these field types in the standard avo-fields.

**Comment by Paul-Bob** [comment]
Hello @TobiasKampAdmin thanks for your feedback.

We are still working on improving the custom dynamic filters DSL.

The DSL that you shared on the code snippet is obsolete.

In the documentation, you'll find some snippets that show how to set an association filter and other useful use cases.

The `select` filter type only works when bound to a field, API to pass options into it is still a work in progress.

Since we are still adapting the filters would be useful to get a list of some concrete use cases so we can adapt to them.

**Comment by Paul-Bob** [comment]
I am currently enhancing the documentation, I'll come back with a more comprehensive answer when is complete.

**Comment by Paul-Bob** [comment]
> In the release video you mentioned there are 2 types (`text` and `date`) to use when you define the dynamic_filters yourself like in the snippet below. But in the documentation it is confusing that other typed like `boolean` or `select` are also possible: [ref](https://docs.avohq.io/3.0/dynamic-filters.html#type).

I get the confusion, there are more than 2 filter types, and they are specified on the [documentation](https://docs.avohq.io/3.0/dynamic-filters.html#type).

> Snippet quote:
```ruby
  self.dynamic_filters = {
    filters: [
      {
        label: "Firstname (custom)",
        conditions:{
          case_sensitive: "Case sensitsive",
          case_insensitive: "Case insensitve"
        }.invert,
        type: :text,  # what are the options here and how to configure?
        icon: "arrow-up-tray",
        query: -> {
          case filter_param.condition.to_sym
          when :case_sensitive
            query.where("firstname LIKE ?", "%#{filter_param.value}%")
          when :case_insensitive
            query.where("firstname ILIKE ?", "%#{filter_param.value}%")
          end 
        }
      }
    ]
  }
```

This DSL remained on the PR description by mistake, it is not used, the same result can be achieved by:

```ruby
def filters
  dynamic_filter label: "Firstname (custom)",
    conditions:{
      case_sensitive: "Case sensitsive",
      case_insensitive: "Case insensitve"
    }.invert,
    type: :text,  # what are the options here and how to configure?
    icon: "arrow-up-tray",
    query: -> {
      case filter_param.condition.to_sym
      when :case_sensitive
        query.where("firstname LIKE ?", "%#{filter_param.value}%")
      when :case_insensitive
        query.where("firstname ILIKE ?", "%#{filter_param.value}%")
      end
    }
end
```
 
Let me know if something misses from [custom dynamic filters](http://localhost:3011/3.0/dynamic-filters.html#custom-dynamic-filters) documentation.

> Could you provide some more information on what the options are at the moment?
When all of these options would be possible, that would be perfect for us!

These are the available options for now:

- [label](https://docs.avohq.io/3.0/dynamic-filters.html#label)
- [icon](https://docs.avohq.io/3.0/dynamic-filters.html#icon)
- [type](https://docs.avohq.io/3.0/dynamic-filters.html#type)
- [query](https://docs.avohq.io/3.0/dynamic-filters.html#query)
- [conditions](https://docs.avohq.io/3.0/dynamic-filters.html#conditions)
- [query_attributes](https://docs.avohq.io/3.0/dynamic-filters.html#query_attributes)
- [suggestions](https://docs.avohq.io/3.0/dynamic-filters.html#suggestions)

> If this is or will become possible, we would like some more information/documentation about the has_many, select and tags types. How to load the possible values, enforce these suggestions...
Basically all you can do in these field types in the standard avo-fields.

Some filter types still lack DSL support for certain options, such as `options` for select filters or `enforce_suggestions` for tag filters. We are working on the DSL, and as previously mentioned, any real-world example of use cases would provide valuable input for us to adapt the DSL.

**Comment by pjmuller** [comment]
@Paul-Bob all clear, we will create follow up github issues (like https://github.com/avo-hq/avo/issues/3010 ) where we will specify which DSL is not working yet. We'll close this one
